src/particleRepresentation/particle/Particle.cpp

```diff
@@ -37,8 +51,66 @@ double Particle::calculateEKin() const {
     return 0.5 * m * (v[0] * v[0] + v[1] * v[1] + v[2] * v[2]);
 }
 
+double Particle::calculateEKinFlow(std::array<double, 3> &avgVelocity) const {
+    auto vHat = v - avgVelocity;
+    return 0.5 * m * (vHat[0] * vHat[0] + vHat[1] * vHat[1] + vHat[2] * vHat[2]);
+}
+
+void Particle::addDirectNeighbor(int idToAdd) {
+    directNeighbors.push_back(idToAdd);
+}
+
+void Particle::addDiagonalNeighbor(int idToAdd) {
+    diagonalNeighbors.push_back(idToAdd);
+}
+
+bool Particle::isDirectNeighbor(Particle &neighbor) {
+    //Check, if p points at neighbor, or neighbor points at p
+    return std::find(directNeighbors.begin(), directNeighbors.end(), neighbor.getId()) != directNeighbors.end()
+           or std::find(neighbor.directNeighbors.begin(), neighbor.directNeighbors.end(), id) != neighbor.
+           directNeighbors.end();
```

Comment:
You could look at parallelizing the [std::find](https://en.cppreference.com/w/cpp/experimental/parallelism/existing)